samples/scripts/build-logs

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+
+if [ $# -eq 1 ]; then
+    BUILD_RUN_NAME=$1
+    FOLLOW=0
+elif [ $# -eq 2 ] && [ "$2" = "--follow" ]; then
+    BUILD_RUN_NAME=$1
+    FOLLOW=1
+else
+    echo "Usage: build-logs <BUILD_RUN_NAME> [--follow]"
+    exit 1
+fi
+
+BUILD_RUN_NAME=$1
+
+# Retrieve the buildrun
+
+BUILD_RUN=`kubectl get buildrun ${BUILD_RUN_NAME} -o json 2>/dev/null`
+
+if [ $? -ne 0 ]; then
+    echo "A build run with the name '${BUILD_RUN_NAME}' cannot be found."
+    exit 2
+fi
+
+echo "Build run: ${BUILD_RUN_NAME}"
+
+# Extract the task run name from the buildrun
+
+set -e
+
+TASK_RUN_NAME=`echo ${BUILD_RUN} | jq -r '.status.latestTaskRunRef'`
+
+echo "Task run: ${TASK_RUN_NAME}"
+
+# Find the pod, it has labels for the build run and task run name
+
+PODS=`kubectl get pods -l "buildrun.build.dev/name=${BUILD_RUN_NAME},tekton.dev/taskRun=${TASK_RUN_NAME}" -o json`
+PODS_LENGTH=`echo ${PODS} | jq ".items | length"`
+
+if [ ${PODS_LENGTH} -eq 0 ]; then
+    echo "No pod found. There is probably a problem in your build configuration."
+    exit 3
+elif [ ${PODS_LENGTH} -ne 1 ]; then
+    echo "More than one pod found, that's unexpected."
+    exit 4
+fi
+
+POD=`echo ${PODS} | jq ".items[0]"`
+POD_NAME=`echo ${POD} | jq -r ".metadata.name"`
+
+echo "Pod: ${POD_NAME}"
+echo
+
+# Retrieve the number of containers
+
+CONTAINERS_LENGTH=`echo ${POD} | jq ".spec.containers | length"`
+
+# Iterate the containers
+
+for (( i = 0; i < ${CONTAINERS_LENGTH}; i++ ))
+do
+    # Extract the container name
+
+    CONTAINER_NAME=`echo ${POD} | jq -r ".spec.containers[${i}].name"`
+    echo "Logs of container ${CONTAINER_NAME}:"
+    echo
+
+    # Check if the container is waiting
+
+    WAITING_REASON=`echo ${POD} | jq -r ".status.containerStatuses[${i}].state.waiting.reason"`
+
+    if [ "${WAITING_REASON}" != "null" ]; then
+        # In follow mode, wait for the container not to be waiting anymore, otherwise stop here
+        if [ ${FOLLOW} = 1 ]; then
+            while [ "${WAITING_REASON}" != "null" ]; do
+                echo "Container is not yet running. Waiting reason: ${WAITING_REASON}"
+                sleep 1
+
+                POD=`kubectl get pod ${POD_NAME} -o json`
+                WAITING_REASON=`echo ${POD} | jq -r ".status.containerStatuses[${i}].state.waiting.reason"`
+            done
+        else
+            echo "Container is not yet running. Waiting reason: ${WAITING_REASON}"
+            echo
+            break
+        fi
+    fi
+
+    # Extract the exit code of the container
+
+    EXIT_CODE=`echo ${POD} | jq ".status.containerStatuses[${i}].state.terminated.exitCode"`
+
+    if [ ${FOLLOW} = 1 ] && [ "${EXIT_CODE}" == "null" ]; then
+        # Container is still running and follow logs is requested
+        kubectl logs -f ${POD_NAME} ${CONTAINER_NAME}
```

Comment:
double quote to prevent globbing `kubectl logs -f "${POD_NAME}" "${CONTAINER_NAME}"`

Reply:
Done.